nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-5-processors/src/main/java/org/apache/nifi/processors/elasticsearch/JsonQueryElasticsearch5.java

```diff
@@ -0,0 +1,291 @@
+            .name("el5-split-up-aggregations")
+            .displayName("Split up aggregation results")
+            .description("Split up aggregation results into one flowfile per result.")
+            .allowableValues(SPLIT_UP_HITS_NO, SPLIT_UP_YES)
+            .defaultValue(SPLIT_UP_HITS_NO.getValue())
+            .required(true)
+            .expressionLanguageSupported(false)
+            .build();
+
+    public static final PropertyDescriptor CLIENT_SERVICE = new PropertyDescriptor.Builder()
+            .name("el5-client-service")
+            .displayName("Client Service")
+            .description("An ElasticSearch client service to use for running queries.")
+            .identifiesControllerService(ElasticSearchClientService.class)
+            .required(true)
+            .build();
+
+    private static final Set<Relationship> relationships;
+    private static final List<PropertyDescriptor> propertyDescriptors;
+
+    private ElasticSearchClientService clientService;
+
+    static {
+        final Set<Relationship> _rels = new HashSet<>();
+        _rels.add(REL_ORIGINAL);
+        _rels.add(REL_FAILURE);
+        _rels.add(REL_HITS);
+        _rels.add(REL_AGGREGATIONS);
+        relationships = Collections.unmodifiableSet(_rels);
+
+        final List<PropertyDescriptor> descriptors = new ArrayList<>();
+        descriptors.add(QUERY);
+        descriptors.add(INDEX);
+        descriptors.add(TYPE);
+        descriptors.add(CLIENT_SERVICE);
+        descriptors.add(SPLIT_UP_HITS);
+        descriptors.add(SPLIT_UP_AGGREGATIONS);
+
+        propertyDescriptors = Collections.unmodifiableList(descriptors);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propertyDescriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        clientService = context.getProperty(CLIENT_SERVICE).asControllerService(ElasticSearchClientService.class);
+    }
+
+    @OnUnscheduled
+    public void onUnscheduled() {
+        this.clientService = null;
+    }
+
+
+    private ObjectMapper mapper = new ObjectMapper();
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        try {
+
+            String query = context.getProperty(QUERY).evaluateAttributeExpressions(flowFile).getValue();
+            String index = context.getProperty(INDEX).evaluateAttributeExpressions(flowFile).getValue();
+            String type = context.getProperty(TYPE).evaluateAttributeExpressions(flowFile).getValue();
+
+            Optional<ElasticSearchResponse> resp = clientService.search(query, index, type);
+            ElasticSearchResponse response = resp.get();
```

Comment:
the search() method returns an Optional, any chance the response object might not be present? Perhaps check isPresent() and send any incoming (not created) flow file to failure?